client/my-sites/patterns/lib/get-pattern-permalink.ts

```diff
@@ -19,3 +19,14 @@ export function getPatternPermalink(
 	url.hash = `#pattern-${ pattern.ID }`;
 	return url.toString();
 }
+
+export const extractPatternIdFromHash = () => {
```

Comment:
Nit. I would probably move this to its own file, since the current setup is such that the file name matches the function name. But I see that there's been a few suggestions on where to place this function in previous reviews, which is why we landed the function here now. I'll leave it up to you to decide.